src/KeyEntityFormEnhancer.php

```diff
@@ -452,8 +465,19 @@ private function createSuggestion(\Exception $exception, KeyInterface $key): Mar
     /** @var \Drupal\apigee_edge\Plugin\KeyType\ApigeeAuthKeyType $key_type */
     $key_type = $key->getKeyType();
 
+    if ($exception instanceof AuthenticationKeyException) {
+      $suggestion = $this->t('@fail_text Verify the Apigee Edge connection settings.', [
+        '@fail_text' => $fail_text,
+      ]);
+    }
+    // Invalid key / OpenSSL unable to sign data.
+    elseif ($exception instanceof DomainException) {
```

Comment:
This should be captured and handled in the API client level first: https://github.com/apigee/apigee-client-php/pull/71/files/3a50936a4198dfaf9377f674725ba59e74f3234e#r341003440 The client should return a `HybridOauth2AuthenticationException` and the previous exception should be the `DomainException` that you can capture here if you want.

Reply:
Done :wink: